org.mwc.cmap.legacy/src/MWC/GUI/Undo/UndoBuffer.java

```diff
@@ -86,6 +86,7 @@
 {
   static protected final int undo = 1;
   static protected final int redo = 2;
+  private static final int UNSET = -1;
   private final Vector<Action> theActions;
   private int presentAction;

```

Comment:
@ayeshare - you should also initialise `presentAction` to `UNSET`

Reply:
done